pulp_ansible/app/sigstore/tasks/sigstore_signature.py

```diff
@@ -221,11 +205,9 @@ async def run(self):
         msg = _("Signing new CollectionVersions with Sigstore.")
         async with ProgressReport(message=msg, code="sign.new.signature", total=ntotal) as p:
             self.progress_report = p
-            await asyncio.create_task(
-                self.sigstore_sign_collection_versions(
-                    sync_to_async_iterable(new_content.iterator())
-                )
-            )
+            async for collection_version in sync_to_async_iterable(new_content.iterator()):
+                tasks.append(asyncio.create_task(self.sigstore_sign_collection_version(collection_version)))
+                await asyncio.gather(*tasks)
```

Comment:
```suggestion
            await asyncio.gather(*tasks)
```
This needs to be outside the for loop else we would still be doing the tasks serially.